File: lib/models/global_product.dart
class GlobalProduct {
  const GlobalProduct({
    required this.id,
    required this.name,
    this.description,
    this.barcode,
    this.unit = 'pcs',
    this.globalCategoryId,
    this.categoryName,
  });

  final int id;
  final String name;
  final String? description;
  final String? barcode;
  final String unit;
  final int? globalCategoryId;
  final String? categoryName;

  factory GlobalProduct.fromJson(Map<String, dynamic> json) {
    return GlobalProduct(
      id: json['id'] as int,
      name: json['name'] as String? ?? '',
      description: json['description'] as String?,
      barcode: json['barcode'] as String?,
      unit: json['unit'] as String? ?? 'pcs',
      globalCategoryId: json['global_category_id'] as int?,
      categoryName: json['category_name'] as String?,
    );
  }
}

File: services/api/app/Models/GlobalProduct.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class GlobalProduct extends Model
{
    protected $fillable = [
        'global_category_id',
        'name',
        'description',
        'barcode',
        'unit',
    ];

    public function category(): BelongsTo
    {
        return $this->belongsTo(GlobalCategory::class, 'global_category_id');
    }

    public function businessProducts(): HasMany
    {
        return $this->hasMany(Product::class, 'global_product_id');
    }
}
